src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -177,8 +177,18 @@ internal void DownloadRecommendationFiles(HashSet<string> allReferences)
                     }
                     catch (Exception)
                     {
-                        //We are checking which files have a recommendation, some of them won't
-                        SkipDownloadFiles.TryAdd(fullFileName, false);
+                        try
+                        {
+                            // Check Beta Recommendations
+                            var fileContents = httpClient.GetStringAsync(string.Concat(Constants.S3BetaRecommendationsBucketUrl, "/", fileName)).Result;
+                            File.WriteAllText(fullFileName, fileContents);
+                        }
+
+                        catch (Exception)
+                        {
+                            //We are checking which files have a recommendation, some of them won't
+                            SkipDownloadFiles.TryAdd(fullFileName, false);
```

Comment:
Why are we adding it only in an error case? Do we need to redownload it every time?

Reply:
Moved try catch logic inside the File Download, that way we can check on File Exists